ucfg.go

```diff
@@ -132,6 +133,46 @@ func (c *Config) Parent() *Config {
 	}
 }
 
+// FlattenedKeys return a flattened views of the set keys in the configuration
+func (c *Config) FlattenedKeys(opts ...Option) []string {
+	var keys []string
+	normalizedOptions := makeOptions(opts)
+
+	if c.IsDict() {
+		for _, v := range c.fields.dict() {
+
+			subcfg, err := v.toConfig(normalizedOptions)
+			if err != nil {
+				ctx := v.Context()
+				p := ctx.path(normalizedOptions.pathSep)
+				keys = append(keys, p)
+			} else {
+				newKeys := subcfg.FlattenedKeys(opts...)
+				keys = append(keys, newKeys...)
+			}
+		}
+	} else if c.IsArray() {
+		for _, a := range c.fields.array() {
+			scfg, err := a.toConfig(normalizedOptions)
+
+			if err != nil {
+				ctx := a.Context()
+				p := ctx.path(normalizedOptions.pathSep)
+				keys = append(keys, p)
+			} else {
+				newKeys := scfg.FlattenedKeys(opts...)
+				keys = append(keys, newKeys...)
+			}
+		}
+	} else {
+		// We should not get here
+		panic("unknown config type")
+	}
+
+	sort.Strings(keys)
```

Comment:
We need to sort here? If so the godoc should reflect the list returned is always being sorted.